code/game/objects/obj_defense.dm

```diff
@@ -0,0 +1,137 @@
+//We will round to this value in damage calculations.
+#define DAMAGE_PRECISION 0.1
+
+//the essential proc to call when an obj must receive damage of any kind.
+/obj/proc/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir, armour_penetration = 0)
```

Comment:
Could you turn this into `take_damage_obj()` or similar if you are keeping the name?
Easier to search and replace for different kinds of damage procs that way,